src/KingSystem/Map/mapObject.cpp

```diff
@@ -1,7 +1,373 @@
+
+    if (dist > 0.0)
+        return dist * mult;
+    else
+        return 100.0 * mult;
+}
+
+sead::Vector3f Object::getScale() const {
+    sead::Vector3f vec;
+
+    mMubinIter.getScale(&vec);
+    return vec;
+}
+
+const char* Object::getUnitConfigName() const {
+    auto* mgr = PlacementMgr::instance();
+
+    if (mgr != nullptr && mgr->mPlacementActors != nullptr)
+        return mgr->mPlacementActors->mActorData[mActorDataIdx].mActorName.cstr();
+
+    return getUnitConfigNameFromByaml();
+}
+
+const char* Object::getUnitConfigNameFromByaml() const {
+    const char* result;
+    if (!mMubinIter.tryGetParamStringByKey(&result, "UnitConfigName"))
+        result = "";
+
+    return result;
+}
+
+Object* Object::findPlacementLODLinkObject(const PlacementActors*) const {
+    if (!mLinkData)
+        return nullptr;
+
+    ObjectLink* result = mLinkData->mLinksToSelf.findLinkWithType(MapLinkDefType::PlacementLOD);
+    if (result)
+        return result->other_obj;
+
+    return nullptr;
+}
+
+Object* Object::findPlacementLODLinkObject() const {
+    if (!mLinkData)
+        return nullptr;
+
+    ObjectLink* result = mLinkData->mLinksToSelf.findLinkWithType(MapLinkDefType::PlacementLOD);
+    if (result)
+        return result->other_obj;
+
+    return nullptr;
+}
+
+const char* Object::getHashIdStringDebug() const {
+    act::ActorDebug::HashUnused t;
+
+    const char* terminator;
+
+    if (mMubinIter.tryGetParamUIntByKey(&t.hash, "HashId")) {
+        terminator = act::ActorDebug::instance()->getNullStr(&t);
+    } else
+        terminator = "";
+
+    return terminator;
+}
+
+const char* Object::getHashIdStringDebug_0() const {
+    act::ActorDebug::HashUnused t;
+
+    const char* terminator;
+
+    if (mMubinIter.tryGetParamUIntByKey(&t.hash, "HashId")) {
+        terminator = act::ActorDebug::instance()->getNullStr(&t);
+    } else
+        terminator = "";
+
+    return terminator;
+}
+
+void* Object::getRails() const {
+    if (!mLinkData)
+        return nullptr;
+
+    return mLinkData->mRails;
+}
+
+void* Object::getRails_0() const {
+    if (!mLinkData)
+        return nullptr;
+
+    return mLinkData->mRails;
+}
+
+bool Object::allocLinkData(sead::ExpHeap* heap) {
+    mLinkData = new (heap) ObjectLinkData;
+
+    return mLinkData != nullptr;
+}
+
+bool Object::hasGenGroup() const {
+    if (!mLinkData)
+        return false;
+
+    return mLinkData->mGenGroup != nullptr;
+}
+
+void Object::incrementLinkNum() {
+    if (mNumLinksPointingToMe != 0xFF) {
+        mNumLinksPointingToMe += 1;
+        return;
+    }
+
+    auto name = sead::SafeString(getUnitConfigName());
+
+    if (name == "LinkTagNone") {
+        // More than 255 actors are specified in one generation group.
+        // If it is not a test, consult the programmer.
+        auto msg = sead::SafeString("一つの生成グループに 255 以上のアクタが指定されています。"
+                                    "テストで無ければプログラマに相談");
+        printDebugMsg(this, &msg, nullptr);
```

Comment:
After changing the signature of printDebugMsg (first review comment), this can just be:

```cpp
printDebugMsg(this, "一つの生成グループに 255 以上のアクタが指定されています。"
                    "テストで無ければプログラマに相談", nullptr);
```